### PR TITLE
integrate shift domain in ITKReader

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
By testing your PR I include it in the ITKReader, if it is fine for you the default option for me it is perfect since all the tools did not need any change. But we can change and I will edit the tools ;)

# Checklist

- [ ] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [ ] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [ ] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [ ] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [ ] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/io/readers/ITKReader.h [hunk 1]
```diff
@@ -104,7 +105,7 @@ namespace DGtal
               typename ITKIOTrait<typename TImage::Value>::DefaultReadFunctor>
     static Image importITK(
     const std::string & filename,
-    const TFunctor & aFunctor = TFunctor() );
+    const TFunctor & aFunctor = TFunctor(), bool shiftDomain=true);
```
Comment (phcerdan) [inline src/DGtal/io/readers/ITKReader.h:108]: What do you think about changing the name to `shiftDomainUsingOrigin`
And what about setting the default to false?

File: src/DGtal/io/readers/ITKReader.ih [hunk 2]
```diff
@@ -188,8 +188,17 @@ namespace DGtal {
...
-    const TypeDGtalImage image( itk_image );
-
+    TypeDGtalImage image( itk_image );
+    if (shiftDomain)
+    {
+      auto c = itk_image->GetOrigin();
+      typename Domain::Point p;
+      for (unsigned int k = 0; k < dimension; k++)
+      {
+        p[k]=c[k];
```
Comment (phcerdan) [inline src/DGtal/io/readers/ITKReader.ih:198]: I would put a static_cast here, to be explicit about the change of type.
And the limitations that an origin of 0.5, would not change the domain.

File: src/DGtal/io/readers/ITKReader.ih [hunk 3]
```diff
@@ -188,8 +188,17 @@ namespace DGtal {
       throw IOException();
     }
     
-    const TypeDGtalImage image( itk_image );
-
+    TypeDGtalImage image( itk_image );
+    if (shiftDomain)
+    {
+      auto c = itk_image->GetOrigin();
```
Comment (phcerdan) [inline src/DGtal/io/readers/ITKReader.ih:194]: add const:
`const auto c = itk_image->GetOrigin();`

**Review by phcerdan — COMMENTED** [review]
Great that is working for you, I wonder if it would be better to open a new PR.

See minor comments, and the question about changing the default.

**Comment by kerautret** [comment]
@phcerdan Thanks agree, I will apply another PR and it gives always conflict ;)
Thanks for the suggestions I will integrate it in the upcoming PR ;) You can close this one.

**Pull request closed without merging**